modules/abstract-utxo/src/transaction.ts

```diff
@@ -207,21 +223,57 @@ export function explainPsbt<TNumber extends number | bigint>(
     throw new Error('failed to parse psbt hex');
   }
   const txOutputs = psbt.txOutputs;
-  function getChangeAddresses() {
+
+  function getChainAndIndexFromBip32Derivations(output: bitgo.PsbtOutput) {
+    const derivations = output.bip32Derivation ?? output.tapBip32Derivation ?? undefined;
+    if (!derivations) {
+      return undefined;
+    }
+    const paths = derivations.map((d) => d.path);
+    if (!paths || paths.length !== 3) {
+      throw new Error('expected 3 paths in bip32Derivation or tapBip32Derivation');
+    }
+    if (!paths.every((p) => paths[0] === p)) {
+      throw new Error('expected all paths to be the same');
+    }
+
+    paths.forEach((path) => {
+      if (paths[0] !== path) {
+        throw new Error(
+          'Unable to get a single chain and index on the output because there are different paths for different keys'
+        );
+      }
+    });
+    const derivationIndices = paths[0].split('/');
+    return {
+      chain: Number(derivationIndices[2]),
+      index: Number(derivationIndices[3]),
+    };
```

Comment:
please add a function `getChainAndIndexFromPath(s: string) { chain: number, index: number }` that has more verification:

* check that path has right number of components
* check that chain and index are positive integers

pls add unit tests